src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -8716,8 +8716,8 @@ private static void AssertTupleTypeEquality(TypeSymbol tuple)
                     if (i != j)
                     {
                         Assert.NotSame(members[i], members[j]);
-                        Assert.False(members[i].Equals(members[j]));
-                        Assert.False(members[j].Equals(members[i]));
+                        Assert.True(!members[i].Equals(members[j]) || members[i].Name != members[j].Name);
+                        Assert.True(!members[j].Equals(members[i]) || members[j].Name != members[i].Name);
```

Comment:
When is the second condition necessary? (When is `Equals()` true but `Name` different?)